src/NServiceBus.CustomChecks/TimerBasedPeriodicCheck.cs

```diff
@@ -17,23 +18,47 @@ public TimerBasedPeriodicCheck(ICustomCheck customCheck, ServiceControlBackend s
             this.ttl = ttl;
         }
 
-        public void Start()
+        public Task Start(CancellationToken cancellationToken)
         {
-            timer = new AsyncTimer();
-            timer.Start(Run, customCheck.Interval, e => { /* should not happen */});
+            stopPeriodicChecksTokenSource = new CancellationTokenSource();
+
+            _ = Task.Run(async () =>
+            {
+                if (!customCheck.Interval.HasValue)
+                {
+                    await Run(stopPeriodicChecksTokenSource.Token).ConfigureAwait(false);
+                    return;
+                }
+
+                while (!stopPeriodicChecksTokenSource.IsCancellationRequested)
+                {
+                    await Run(stopPeriodicChecksTokenSource.Token).ConfigureAwait(false);
+
+                    await Task.Delay(customCheck.Interval.Value, stopPeriodicChecksTokenSource.Token).ConfigureAwait(false);
```

Comment:
Without catching OperationCancelledException in the Task.Run block, couldn't this crash the process during shutdown? (Previous line too.)

Reply:
I think these days nothing happens, but I guess that it's better to be safe than sorry?

Reply:
Try catch added